tests/table/test_metadata.py

```diff
@@ -755,3 +755,14 @@ def test_make_metadata_fresh() -> None:
     )
 
     assert actual.model_dump() == expected.model_dump()
+
+
+def test_schema_immutability(example_table_metadata_v2: Dict[str, Any]) -> None:
+    table_metadata = TableMetadataV2(**example_table_metadata_v2)
+
+    # Schema id 1 has 2 columns of decimal and fixed columns
+    # to validate the schema immutability of Singleton types
+    original_schema = table_metadata.schemas[1]
+    copied_schema = table_metadata.model_copy(deep=True).schemas[1]
```

Comment:
nit: instead of checking for `model_copy` behavior here, we can move this to the `test_types.py` with something like this 
https://github.com/apache/iceberg-python/pull/838/files#diff-bd467a363c9f155a2e6a7716aeca0c351089135a53d09a66138cea4f16506ebdR624-R630

Reply:
Thank you for the review @kevinjqliu. 

The `model_copy` with `deep=True` only runs in the update_metadata function. The idea here is to simulate this behavior, but I believe it would be better to move this test to run the `update_metadata` function with `enforce_validation` set to True. This way, we will test what really happens in the real world.

https://github.com/apache/iceberg-python/blob/a6cd0cf325b87b360077bad1d79262611ea64424/pyiceberg/table/__init__.py#L1059-L1081

What do you think?

Reply:
sounds good! 

I think thats the only place where `model_copy(deep=True)` is used in the codebase
https://github.com/search?q=repo%3Aapache%2Ficeberg-python%20model_copy(deep%3DTrue)%20&type=code

I would also suggest creating a new table metadata example instead of `EXAMPLE_TABLE_METADATA_V2`, looks like changing `EXAMPLE_TABLE_METADATA_V2` broke a lot of tests.